src/test/ui/suggestions/args-instead-of-tuple.rs

```diff
@@ -0,0 +1,18 @@
+// Test suggesting tuples where bare arguments may have been passed
+// See issue #86481 for details.
+
+// run-rustfix
+
+fn main() {
+    let _: Result<(i32, i8), ()> = Ok(1, 2);
+    //~^ ERROR this enum variant takes 1 argument but 2 arguments were supplied
+    let _: Option<(i32, i8, &'static str)> = Some(1, 2, "hi");
+    //~^ ERROR this enum variant takes 1 argument but 3 arguments were supplied
+    let _: Option<()> = Some();
```

Comment:
Ah, so the single-field tuple code is handled separately (#91530), that we don't overlap with here. I can add a test as a placeholder that this PR/change doesn't address, for completeness?

Reply:
But your compiler code produces and handles `FnArgsAsTuple::Single` — which are single-field tuples?

Reply:
Yes, I've come to the realisation that I believe this is unreachable. I think this because the code at the moment relies on an argument count mismatch, (e.g. `fn f(_: (T, U)){}` and `f(T, U)`, so 1 != 2) but this will never be the case for 1-tuples. I've deleted the `Single` case and tidied the code, with the 1-tuple being entirely handled by #91530, if this is ok by you? I've left a test in for error/mismatch with 1-tuples, just to cover that base in the meanwhile.

----

A slightly more thorough check to convince myself it's unreachable code:

```rust
fn f(_: (i32,)) {}

fn main() {
    f(1); // 1
    f(("",)); // 2
    f((1, 2)); // 3
    f(); // 4
    f(1, 2); // 5
}
```
1. The argument count matches so we take the below code and just diagnose a tuple vs. primitive type mismatch (handled by #91530) - nothing to parenthesize

https://github.com/rust-lang/rust/blob/bd3cb52565faab2755ff1bdb54d88bc91f47b4b9/compiler/rustc_typeck/src/check/fn_ctxt/checks.rs#L281-L282

2. Another type mismatch, nothing to parenthesize
3. Tuple vs. tuple, nothing to parenthesize
4. Empty vs. tuple - mismatching counts
5. Mismatching counts

Reply:
I don't feel happy with panicking the compiler with this code though, it feels too easy for a future change to expose this. Perhaps it's better we just return `None` instead of the `unreachable!`.

https://github.com/rust-lang/rust/blob/9eade24485628016fa5dcb05508629f3a2bff037/compiler/rustc_typeck/src/check/fn_ctxt/checks.rs#L522-L524

Reply:
As long as you have some tests, I think the `unreachable!` is probably fine since this code is an error path.

Reply:
Great, SGTM :+1: